File: 2026-03/2026-03-25/cooldownTime.js
function canRetake(finishTime, currentTime) {
  const HOURS_IN_DAY = 24;
  const MINUTES_IN_HOUR = 60;
  const SECONDS_IN_MINUTE = 60;
  const EXAM_COOLDOWN_HOURS = 48;

  const finishTimeObject = Temporal.PlainDateTime.from(finishTime);
  const currentTimeObject = Temporal.PlainDateTime.from(currentTime);
  const timeDifference = currentTimeObject.since(finishTimeObject);

  const timeDifferenceHours = 
    timeDifference.days * HOURS_IN_DAY + 
    timeDifference.hours + 
    timeDifference.minutes / MINUTES_IN_HOUR + 
    timeDifference.seconds / SECONDS_IN_MINUTE / MINUTES_IN_HOUR;

  return timeDifferenceHours >= EXAM_COOLDOWN_HOURS ? true : false;
}

// --- TEST SUITE ---

const testsText = String.raw`
1. canRetake("2026-03-23T08:00:00", "2026-03-25T14:00:00") should return true.
2. canRetake("2026-03-24T14:00:00", "2026-03-25T10:00:00") should return false.
3. canRetake("2026-03-23T09:25:00", "2026-03-25T09:25:00") should return true.
4. canRetake("2026-03-23T11:50:00", "2026-03-25T11:49:59") should return false.
`;

const testsRegex = /(?<number>\d+)\.\s(?<functionCall>.+) should return (?<output>.+?)\.?$/gm;
const testData = [...testsText.matchAll(testsRegex).map(match => match.groups)];

function runTests(testData) {
  console.log("--------------------------");
  console.log("🧪Starting Verification...");
  console.log("--------------------------");

  let failCount = 0;
  
  testData.forEach(test => {
    const functionCallOutput = eval(test.functionCall);
    const testOutput = eval(test.output);

    const comparison = Array.isArray(testOutput)
      ? arraysEqual(functionCallOutput, testOutput)
      : functionCallOutput === testOutput;

    if (comparison) {
      console.log(`${test.number}.✅PASS - Function Call: ${test.functionCall}`);
    }
    else {
      console.log(`${test.number}.❌FAIL - Function Call: ${test.functionCall}\nExpected: ${testOutput}\nGot: ${functionCallOutput}`);
      ++failCount;
    }
  })

  console.log("----------------------------",
    failCount
    ? `\n⚠️WARNING: ${failCount}/${testData.length} tests FAILED.`
    : "\n🎉SUCCESS: All tests PASSED."
  );
}

function arraysEqual(a, b) {
  if (a.length !== b.length)
    return false;

  for (let i = 0; i < a.length; ++i) {
    if (Array.isArray(a[i])) {
      if (Array.isArray(b[i])) {
        if (!arraysEqual(a[i], b[i]))
          return false;
      }
      else 
        return false;
    }

    else if (a[i] !== b[i])
      return false;
  }

  return true;
}

runTests(testData);

File: 2026-03/2026-03-25/cooldown_time.py
import datetime as dt

def can_retake(finish_time, current_time):
    HOURS_IN_DAY = 24;
    MINUTES_IN_HOUR = 60;
    SECONDS_IN_MINUTE = 60;
    EXAM_COOLDOWN_HOURS = 48;

    finish_time_object = dt.datetime.fromisoformat(finish_time)
    current_time_object = dt.datetime.fromisoformat(current_time)
    time_difference = current_time_object - finish_time_object

    time_difference_hours = \
        time_difference.days * HOURS_IN_DAY + \
        time_difference.seconds / SECONDS_IN_MINUTE / MINUTES_IN_HOUR
    
    return True if time_difference_hours >= EXAM_COOLDOWN_HOURS else False

# --- TEST SUITE ---

tests_text = r'''
1. can_retake("2026-03-23T08:00:00", "2026-03-25T14:00:00") should return True.
2. can_retake("2026-03-24T14:00:00", "2026-03-25T10:00:00") should return False.
3. can_retake("2026-03-23T09:25:00", "2026-03-25T09:25:00") should return True.
4. can_retake("2026-03-25T11:50:00", "2026-03-23T11:49:59") should return False.
'''

import re

tests_regex = re.compile(r"(?P<number>\d+)\.\s(?P<function_call>.+) should return (?P<output>.+?)\.?$", re.MULTILINE)
test_data = [item.groupdict() for item in re.finditer(tests_regex, tests_text)]

def run_tests(test_data):
    print("--------------------------")
    print("🧪 Starting Verification...")
    print("--------------------------")

    fail_count = 0

    for test in test_data:
        function_call_output = eval(test['function_call'])
        test_output = eval(test['output'])
        
        if function_call_output == test_output:
            print(f"{test['number']}.✅ PASS - Function Call:\n{test['function_call']}")
        else:
            print(f"{test['number']}.❌ FAIL - Function Call:\n{test['function_call']}\nExpected:\n{test_output}\nGot:\n{function_call_output}")
            fail_count += 1

    print("----------------------------",
        f"\n⚠️ WARNING: {fail_count}/{len(test_data)} tests FAILED." if fail_count
        else "\n🎉 SUCCESS: All tests PASSED."
    )

run_tests(test_data)